backend/test_observer/controllers/test_executions/end_test.py

```diff
@@ -0,0 +1,181 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy import select
+from sqlalchemy.orm import Session, joinedload
+
+from test_observer.data_access.models import (
+    Artefact,
+    ArtefactBuild,
+    Stage,
+    TestCase,
+    TestExecution,
+    TestResult,
+)
+from test_observer.data_access.models_enums import (
+    FamilyName,
+    TestExecutionReviewDecision,
+    TestExecutionStatus,
+    TestResultStatus,
+)
+from test_observer.data_access.repository import get_or_create
+from test_observer.data_access.setup import get_db
+
+from .logic import delete_previous_results
+from .models import C3TestResult, C3TestResultStatus, EndTestExecutionRequest
+
+router = APIRouter()
+
+
+@router.put("/end-test")
```

Comment:
We seem to start making a hierarchy of the API endpoints. I remember FastAPI has a support for [tags](https://fastapi.tiangolo.com/tutorial/path-operation-configuration/#tags) which let you split the docs view in separate parts, maybe we can tag these endpoints with `tags=["test-execution"]` to logically separate them in different parts and make the docs a little bit more organized.